### ISSUE TITLE
Map all buttons and controls to Keyboard keys

### ISSUE THREAD

**Issue by kwadroke** [smcameron/space-nerds-in-space#5]
I'm planning on building some custom keyboards for the game and I'd like to be able to control everything in game from the keyboard without having to use a mouse. (These will get double use with Artemis SBS). A config file would be nice to be able to remap keys for future upgrades to code and to my custom keyboards.

**Comment by heaventwig** [comment]
I have some ideas for acidúlalo keyaction names (and associated in-game effects). Where/how might you like to find those listed for possible future development? Maybe on this thread? Maybe as a separate issue report, like "this thing doesn't appear to exist"?
(Examples include things like "select next engineering system" "select previous engineering system" "increase coolant to system" "increase power to system" "raise phaser frequency" "roll left" "cycle science screen selection to next nearer/further object" "science select nearest object" and so on.)

**Comment by smcameron** [comment]
Yeah, this issue is fine.  Or open new issue(s), whatever makes the most sense to you.

**Comment by heaventwig** [comment]
Quick note that my code review just now shows that this file (keymap.txt) now has format like
map station keyname actionname
and I don't know yet what permitted stations are.

**Comment by smcameron** [comment]
I notice the code to read this file does not follow the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html):

 ```
        homedir = getenv("HOME");
        if (homedir == NULL)
                return;

        sprintf(filename, "%s/.space-nerds-in-space/snis-keymap.txt", homedir);
        f = fopen(filename, "r");
        if (!f)
                return;
```

I should fix that.

**Comment by smcameron** [comment]
Committed 578f202da162c3716cfda2f4768ff24065c962d2  Follow XDG directory spec for snis-keymap.txt
